cardano-diffusion/protocols/lib/Cardano/Network/Protocol/ChainSync/Codec/TimeLimits.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE GADTs               #-}
+{-# LANGUAGE PolyKinds           #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.Network.Protocol.ChainSync.Codec.TimeLimits
+  ( timeLimitsChainSync
+  ) where
+
+import Control.Monad.Class.MonadTime.SI
+
+import Network.TypedProtocol.Codec.CBOR hiding (decode, encode)
+
+import Ouroboros.Network.Protocol.ChainSync.Codec
+import Ouroboros.Network.Protocol.ChainSync.Type
+import Ouroboros.Network.Protocol.Limits
+
+import Cardano.Network.PeerSelection.PeerTrustable (PeerTrustable)
+import Cardano.Network.Protocol.Limits
+
+import Data.Bifunctor (first)
+import Data.Kind (Type)
+import System.Random (StdGen, randomR)
+
+-- | Time Limits
+--
+-- +----------------------------+-------------------------------------------------------------+
+-- | ChainSync State            | timeout (s)                                                 |
+-- +============================+=============================================================+
+-- | @'StIdle'@                 | 'waitForever' (i.e. never times out)                        |
+-- +----------------------------+-------------------------------------------------------------+
+-- | @'StNext' 'StCanAwait'@    | 'shortWait'                                                 |
+-- +----------------------------+-------------------------------------------------------------+
+-- | @'StNext' 'StMustReply'@   | randomly picked using uniform distribution from             |
+-- |                            | the range @('minChainSyncTimeout', 'maxChainSyncTimeout')@, |
+-- |                            | which corresponds to a chance of an empty streak of slots   |
+-- |                            | between `0.0001%` and `1%` probability.                     |
+-- +----------------------------+-------------------------------------------------------------+
+-- | @'StIntersect'@            | 'shortWait'                                                 |
+-- +----------------------------+-------------------------------------------------------------+
+--
+timeLimitsChainSync :: forall (header :: Type) (point :: Type) (tip :: Type).
+                       ChainSyncIdleTimeout
+                    -- ^ idle timeout, the default value
+                    -- `Configuration.defaultChainSyncIdleTimeout`.
+                    -> ProtocolTimeLimitsWithRnd (ChainSync header point tip)
```

Comment:
```suggestion
timeLimitsChainSync :: forall (header :: Type) (point :: Type) (tip :: Type).
                       ChainSyncIdleTimeout
                    -- ^ idle timeout, the default value
                    -- `Configuration.defaultChainSyncIdleTimeout`.
                    -> PeerTrustable
                    -> ProtocolTimeLimitsWithRnd (ChainSync header point tip)
```
Instead of modifying `ProtocolTimeLimitsWithRnd`, provide `PeerTrustable` as an argument.